pypsa/linopt.py

```diff
@@ -601,6 +601,175 @@ def set_int_index(ser):
+                        solver_options, warmstart=None, store_basis=True):
+    """
+   Highs solver function. Reads a linear problem file and passes it to the highs
+    solver. If the solution is feasible the function returns the objective,
+    solution and dual constraint variables. Highs must be installed for usage.
+    Documentation: https://www.maths.ed.ac.uk/hall/HiGHS/
+
+    Installation
+    -------------
+    The script might only work for version HiGHS 1.1.1. Installation steps::
+        sudo apt-get install cmake  # if not installed
+        git clone git@github.com:ERGO-Code/HiGHS.git
+        cd HiGHS
+        git checkout 95342daa73543cc21e5b27db3e0fbf7330007541 # moves to HiGHS 1.1.1
+        mkdir build
+        cd build
+        cmake ..
+        make
+        ctest
+
+    Then in .bashrc add paths of executables and library ::
+        export PATH="${PATH}:/foo/HiGHS/build/bin"
+        export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/foo/HiGHS/build/lib"
+        source .bashrc
+
+    Now when typing ``highs`` in the terminal you should see something like ::
+        Running HiGHS 1.1.1 [date: 2021-11-14, git hash: 95342daa]
+
+    Architecture
+    -------------
+    The function reads and execute (i.e. subprocess.Popen,...) terminal
+    commands of the solver. Meaning the command can be also executed at your
+    command window/terminal if HiGHs is installed. Executing the commands on
+    your local terminal helps to identify the raw outputs that are useful for
+    developing the interface further.
+
+    All functions below the "process = ..." do only read and save the outputs
+    generated from the HiGHS solver. These parts are solver specific and
+    depends on the solver output.
+
+    Solver options
+    ---------------
+    Solver options are read by the 1) command window and the 2) option_file.txt
+
+    1) An example list of solver options executable by the command window is given here:
+    Examples:
+    --model_file arg 	File of model to solve.
+    --presolve arg 	    Presolve: "choose" by default - "on"/"off" are alternatives.
+    --solver arg 	    Solver: "choose" by default - "simplex"/"ipm" are alternatives.
+    --parallel arg 	    Parallel solve: "choose" by default - "on"/"off" are alternatives.
+    --time_limit arg 	Run time limit (double).
+    --options_file arg 	File containing HiGHS options.
+    -h, --help 	        Print help.
+
+    2) The options_file.txt gives some more options, see a full list here: 
+    https://www.maths.ed.ac.uk/hall/HiGHS/HighsOptions.set 
+    By default, we insert a couple of options for the ipm solver. The dictionary
+    can be overwritten by simply giving the new values. For instance in PyPSA-Eur,
+    you could write a dictionary replacing some of the default values or adding new
+    options:
+    ```
+    solving:
+        solver:
+            name: highs,
+            method: ipm,
+            parallel: "on",
+            <option_name>: <value>,
+    ```
```

Comment:
This explanation should not refer to PyPSA-Eur. Write how you can use the `solver_options` keyword.

```py
solver_options = {
    "method": "ipm",
    ...
}
```